platform/mbed_watchdog_mgr.cpp

```diff
@@ -0,0 +1,112 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "mbed_watchdog_mgr.h"
+
+static bool is_watchdog_started = false; //boolean to control watchdog start and stop
+#define MS_TO_US(x) ((x) * 1000) //macro to convert millisecond to microsecond
+static uint32_t elapsed_ms = (HW_WATCHDOG_TIMEOUT / 2);
+MBED_STATIC_ASSERT((HW_WATCHDOG_TIMEOUT > 0), "Timeout must be greater than zero");
+
+/** mbed watchdog manager creates watchdog class instance with zero timeout
+ *  as mbed watchdog manager is not going to register(not going to call "start" method of Watchdog class)
+ *  its only going to call process to verify all registered users/threads in alive state
+ *
+ */
+mbed::Watchdog watchdog(0, "Platform Watchdog");
+/** Create singleton instance of LowPowerTicker for watchdog periodic call back of kick.
+ */
+static mbed::LowPowerTicker *get_ticker()
+{
+    static mbed::LowPowerTicker ticker;
```

Comment:
When is this static instance constructed ?
I wonder if it might not cause some race condition/constructor order issue at some point in the future.
Maybe using a `SingletonPtr<mbed::LowPowerTicker>` would solve this problem.

Reply:
@ithinuel `static` variables local to a function are instantiated at first use. `__cxa_guard_acquire`, `__cxa_guard_release` and `__cxa_guard_abort` (see [here](https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_retarget.cpp#L1436-L1469)) are here to protect the instantiation from race condition. 

You can consider SingletonPtr like _syntactic sugar_; it allows you to write in one line of code what is otherwise written in 5.

Reply:
Changed into singletonptr